builtin/providers/azurerm/resource_arm_local_network_gateway.go

```diff
@@ -0,0 +1,158 @@
+package azurerm
+
+import (
+	"fmt"
+	"log"
+
+	"github.com/Azure/azure-sdk-for-go/arm/network"
+	"github.com/Azure/azure-sdk-for-go/core/http"
+	"github.com/hashicorp/terraform/helper/schema"
+)
+
+// resourceArmLocalNetworkGateway returns the schema.Resource
+// associated to an Azure local network gateway.
+func resourceArmLocalNetworkGateway() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceArmLocalNetworkGatewayCreate,
+		Read:   resourceArmLocalNetworkGatewayRead,
+		Update: resourceArmLocalNetworkGatewayUpdate,
+		Delete: resourceArmLocalNetworkGatewayDelete,
+
+		Schema: map[string]*schema.Schema{
+			"name": &schema.Schema{
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+
+			"location": &schema.Schema{
+				Type:      schema.TypeString,
+				Optional:  true,
+				ForceNew:  true,
+				StateFunc: azureRMNormalizeLocation,
+			},
+
+			"resource_group_name": &schema.Schema{
+				Type:     schema.TypeString,
+				Optional: true,
+				ForceNew: true,
+			},
+
+			"resource_guid": &schema.Schema{
+				Type:     schema.TypeString,
+				Optional: true,
+			},
+
+			"gateway_address": &schema.Schema{
+				Type:     schema.TypeString,
+				Required: true,
+			},
+
+			"address_space": &schema.Schema{
+				Type:     schema.TypeList,
+				Required: true,
+				Elem: &schema.Schema{
+					Type: schema.TypeString,
+				},
+			},
+		},
+	}
+}
+
+// resourceArmLocalNetworkGatewayCreate goes ahead and creates the specified ARM local network gateway.
+func resourceArmLocalNetworkGatewayCreate(d *schema.ResourceData, meta interface{}) error {
+	lnetClient := meta.(*ArmClient).localNetConnClient
+
+	name := d.Get("name").(string)
+	location := d.Get("location").(string)
+	resGroup := d.Get("resource_group_name").(string)
+	ipAddress := d.Get("gateway_address").(string)
+
+	// NOTE: due to the including-but-different relationship between the ASM
+	// and ARM APIs, one may set the following local network gateway type to
+	// "Classic" and basically get an old ASM local network connection through
+	// the ARM API. This functionality is redundant with respect to the old
+	// ASM-based implementation which we already have, so we just use the
+	// new Resource Manager APIs here:
+	typ := "Resource Manager"
+
+	// fetch the 'address_space_prefix'es:
+	prefixes := []string{}
+	for _, pref := range d.Get("addres_space").([]interface{}) {
```

Comment:
This is likely causing the panic mentioned below. Should be `address_space` I think?